docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+  
+- spec.exposedRegistryPath: Stores the path of registry route when using direct migration.
+
+- spec.insecure: If set false, user will need to provide CA bundle for TLS connection to the remote cluster.
+
+- spec.isHostCluster: Specifies if the cluster is host (where the controller is installed) or not. This is a required field.
+
+- spec.refresh: If set True, forces the controller to run a full suite of validations on migcluster.
+
+- spec.restartRestic: An override setting to tell the controller that the source cluster restic needs to be restarted after stage pod creation.
+
+- spec.serviceAccountSecretRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.url: Stores the url of the remote cluster. The field is only required for the source cluster object.
+
+## MigHooks
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigHook
+metadata:
+  generateName: test-
+  name: test-btmv5
+  namespace: openshift-migration
+spec:
+  custom: false
+  image: 'quay.io/konveyor/hook-runner:latest'
+  playbook: >-
+    LSBuYW1lOiBTbGVlcCBmb3IgMTAgc2Vjb25kcyBhbmQgY29udGludWUgd2l0aCBwbGF5CiAgd2FpdF9mb3I6CiAgIadsfwerq0OiAxMA==
+  targetCluster: source
+```
+
+- spec.activeDeadlineSeconds: Specifies the highest amount of time for which the hook will run.
+
+- spec.custom: Specifies whether the hook is a custom Ansible playbook or a pre-built image. This is a required field.
+
+- spec.image: Specifies the image of the hook to be executed. This is a required field.
+
+- spec.playbook: Specifies the contents of the custom Ansible playbook in base64 format, it is used in conjunction with the custom boolean flag.
+
+- spec.targetCluster: Specifies the cluster on which the hook is to be executed. This is a required field.
+
+
+
+## MigMigrations
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigMigration
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migmigration-sample
+  namespace: openshift-migration
+spec:
+  # [!] Set 'canceled: true' will cancel the migration
+  canceled: false
+  # [!] Set 'rollback: true' will rollback the migration's plan
+  rollback: true
+  # [!] Set 'stage: true' to run a 'Stage Migration' and skip quiescing of Pods on the source cluster.
+  stage: false
+  # [!] Set 'quiescePods: true' to scale down Pods on the source cluster after the 'Backup' stage of a migration has finished
+  quiescePods: false
+  # [!] Set 'keepAnnotations: true' to retain labels and annotations applied by the migration
+  keepAnnotations: false
+
+  migPlanRef:
+    name: migplan-sample
+    namespace: openshift-migration
+```
+
+- spec.canceled: Invokes the cancel migration operation, when set to true the migration controller switches to cancel itinerary. This field can be used on-demand to cancel the running migration.
+
+- spec.keepAnnotations: Specifies whether to retain the annotations set by the migration controller or not.
+
```

Comment:
@eriknelson @shubham-pampattiwar 

I thought I commented on keepAnnotations but maybe I did it somewhere else. Why would a user not want to retain annotations? Is this part of rollback? 

Or is there another use case in which the user might not want to keep annotations?

Reply:
@apinnick These annotations are for migration purposes, the client might not want them after the migration is done.